src/branch/context.rs

```diff
@@ -1,41 +1,73 @@
 use std::fs;
 use crate::portable::config::Config;
 use std::path::{Path, PathBuf};
+use crate::connect::Connection;
 use crate::credentials;
 use crate::portable::options::InstanceName;
 use crate::portable::project::{instance_name, stash_path};
 
 pub struct Context {
-    pub project_config: Config,
-    pub project_branch: String,
-    pub branch: String,
+    pub project_config: Option<Config>,
+    pub project_branch: Option<String>,
```

Comment:
I though that we abandoned the concept of "projects having branches" at all.

Right now, I see that we still read "edgedb.toml", property "edgedb.branch" and default to that when "credentials.json" doesn't contain the "database".

Reply:
IIRC we're still keeping that, I'll double check on this